File: 이것이 코딩테스트다/python/3장_숫자 카드 게임.py
'''
n, m = map(int, input().split(' '))
mat=[]
for i in range(0, n):
    tmp = list(map(int, input().split(' ')))
    tmp.sort()
    mat.append(tmp)

maxNum=0
for i in range(0, n):
    if maxNum<mat[i][0]:
        maxNum=mat[i][0]

print(maxNum)
'''

n, m = map(int, input().split(' '))

maxNum=0
for i in range(0, n):
    tmp = list(map(int, input().split(' ')))
    tmp.sort()
    if maxNum<tmp[0]:
        maxNum=tmp[0]

print(maxNum)

File: 이것이 코딩테스트다/c++/3장_숫자 카드 게임.cpp
#include <bits/stdc++.h>
#include <algorithm>
using namespace std;

int main(){
	int n, m;
	cin>>n>>m;
	
	vector< vector<int> > mat;
	
	for(int i=0; i<n; i++){
		vector<int> tmp;
		for(int j=0; j<m; j++){
			int num;
			cin>>num;
			tmp.push_back(num);
		}
		sort(tmp.begin(), tmp.end());
		mat.push_back(tmp);
	}
	
//	for(int i=0; i<n; i++){
//		for(int j=0; j<n; j++){
//			printf("%d ", mat[i][j]);
//		}
//		printf("\n");
//	}
	
	int maxNum=0;
	
	for(int i=0; i<n; i++){
		if(maxNum<mat[i][0]){
			maxNum=mat[i][0];
		}
	}
	
	printf("%d", maxNum);
	
}
